pkg/plugins/install.go

```diff
@@ -1,14 +1,97 @@
 package plugins
 
 import (
+	"encoding/json"
+	"fmt"
+	"sort"
+
 	"get.porter.sh/porter/pkg/pkgmgmt"
+	"get.porter.sh/porter/pkg/portercontext"
 )
 
 type InstallOptions struct {
 	pkgmgmt.InstallOptions
+
+	File string
 }
 
-func (o *InstallOptions) Validate(args []string) error {
+func (o *InstallOptions) Validate(args []string, cxt *portercontext.Context) error {
 	o.PackageType = "plugin"
+	if o.File != "" {
+		if len(args) > 0 {
+			return fmt.Errorf("plugin name should not be specified when --file is provided")
+		}
+
+		if o.URL != "" {
+			return fmt.Errorf("plugin URL should not be specified when --file is provided")
+		}
+
+		if o.Version != "" {
+			return fmt.Errorf("plugin version should not be specified when --file is provided")
+		}
+
+		if _, err := cxt.FileSystem.Stat(o.File); err != nil {
+			return fmt.Errorf("unable to access --file %s: %w", o.File, err)
+		}
+
+		return nil
+	}
+
 	return o.InstallOptions.Validate(args)
 }
+
+// InstallFileOption is the go representation of plugin installation file format.
+type InstallFileOption struct {
+	data map[string]pkgmgmt.InstallOptions
+	keys []string
+}
+
+func (io *InstallFileOption) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	data := map[string]pkgmgmt.InstallOptions{}
+	err := unmarshal(&data)
+	if err != nil {
+		return fmt.Errorf("could not unmarshal into plugins.InstallFileOption: %w", err)
+	}
+	return io.sort(data)
+}
+
+func (io *InstallFileOption) UnmarshalJSON(data []byte) error {
+	configs := map[string]pkgmgmt.InstallOptions{}
+	err := json.Unmarshal(data, &configs)
+	if err != nil {
+		return fmt.Errorf("could not unmarshal into plugins.InstallFileOption: %w", err)
+	}
+
+	return io.sort(configs)
+}
+
+// sort returns InstallOptions in alphabetical order.
```

Comment:
Nit: This sorts the plugins and stores that in the InstallOptions, doesn't return them.